includes/chart.js

```diff
@@ -79,75 +115,108 @@ function addNodesAndLinks(rowValues) {
                 newNode.setAttribute("y", y);
-                newNode.setAttribute("width", template_node.width); // Use the width attribute
-                newNode.setAttribute("height", template_node.height); // Use the height attribute
+                newNode.setAttribute("width", template_node.width);
+                newNode.setAttribute("height", template_node.height);
+
                 if (template_node.rx && template_node.ry) {
-                    newNode.setAttribute("rx", template_node.rx); // Set horizontal radius for rounded corners
-                    newNode.setAttribute("ry", template_node.ry); // Set vertical radius for rounded corners
+                    newNode.setAttribute("rx", template_node.rx);
+                    newNode.setAttribute("ry", template_node.ry);
                 }
             }
 
-            newNode.setAttribute("id", `${row.id}_${template_node.id}`);
-            newNode.setAttribute("fill", colors[row.statementType]);
-            newNode.setAttribute("stroke", "black")
+            // Set common attributes for all shapes
+            newNode.setAttribute("fill", colors[rowObj.statementType]); // Use rowObj.statementType to get the correct color
+            newNode.setAttribute("stroke", "lightgray");
+            newNode.setAttribute("stroke-width", "2");
+            newNode.setAttribute("id", `${row[0]}_${template_node.id}`);
+
+            // Append the new node to the nodes group
+            nodesGroup.appendChild(newNode);
 
-            // Add text inside the shape
-            var text = document.createElementNS("http://www.w3.org/2000/svg", "text");
-            var textContent = row[template_node.type];
-            if (textContent.length >= 17) {
-                textContent = textContent.slice(0, 14) + "...";
-            }
-            text.textContent = textContent;
-            text.setAttribute("fill", "black");
-            text.setAttribute("font-size", "12px");
-            text.setAttribute("font-family", "calibri");
-            text.setAttribute("text-anchor", "middle"); // Center align the text
 
+            // Add click event to display the clicked node
+            newNode.addEventListener('click', function () {
+                // Update clicked node text content
+                document.getElementById("clicked-node").textContent = "Clicked Node: " + `${row[0]}_${template_node.id}`;
+
+                // Get the current stroke color
+                let currentStrokeColor = newNode.getAttribute("stroke");
+
+                // Toggle stroke color
+                if (currentStrokeColor !== "green") {
+                    // Change stroke color to light green
+                    newNode.setAttribute("stroke", "green");
+                } else {
+                    // Restore the original stroke color (assuming it was lightgray initially)
+                    newNode.setAttribute("stroke", "lightgray");
+                }
+            });
```

Comment:
Since we don't do anything else with a particular node when it's clicked right now, can we remove this?